src/test/java/org/broadinstitute/hellbender/utils/reference/FastaReferenceWriterUnitTest.java

```diff
@@ -0,0 +1,491 @@
+package org.broadinstitute.hellbender.utils.reference;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.reference.FastaSequenceIndex;
+import htsjdk.samtools.reference.IndexedFastaSequenceFile;
+import htsjdk.samtools.reference.ReferenceSequence;
+import htsjdk.samtools.reference.ReferenceSequenceFileFactory;
+import htsjdk.samtools.util.SequenceUtil;
+import org.broadinstitute.hellbender.engine.ReadsDataSource;
+import org.broadinstitute.hellbender.utils.RandomDNA;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Unit tests for {@link FastaReferenceWriter}.
+ */
+public class FastaReferenceWriterUnitTest extends BaseTest {
+
+    @Test(expectedExceptions = IllegalStateException.class)
+    public void testEmptySequence() throws IOException {
+        final File testOutput = createTempFile("fwr-test", ".fasta");
+        Assert.assertTrue(testOutput.delete());
+        try (final FastaReferenceWriter writer = new FastaReferenceWriter(testOutput.toPath(), false, false)) {
+            writer.startSequence("seq1");
+            writer.appendBases(new RandomDNA(113).nextBases(100));
+            writer.startSequence("seq2");
+            writer.startSequence("seq3");
+        } finally {
+            Assert.assertTrue(testOutput.delete());
```

Comment:
won't testOutput be automatically deleted on shutdown?

Reply:
Leave it as it is; don't see the harm.